### ISSUE TITLE
Thousands of empty guest wishlists being stored

### ISSUE THREAD

**Issue by j-greig** [verbb/wishlist#24]
We have new guest wishlists being created every few minutes, but they are all empty.

Is there a way to prevent this?

(I realise that maybe it's due to our templating approach / setup being wrong)

EDIT: Have just turned on `purgeInactiveLists` via the PHP config file...

![image](https://user-images.githubusercontent.com/616194/59781274-66064900-92b3-11e9-9c14-b1cf62aeea6c.png)

**Comment by j-greig** [comment]
Still stuck with this, any pointers most welcome, thanks!

EDIT: have set `'purgeInactiveListsDuration' => 'P2D',` instead of 8 days, will this help?

**Comment by tidy** [comment]
That change should purge empty lists that haven't been updated in the last 2 days rather than the last 8. However, I think you need to visit the Wishlist screen in the CMS for this purge to trigger?

**Comment by j-greig** [comment]
Ah, didn't realise you had to visit that page to trigger, thank you @tidy !

**Comment by tidy** [comment]
I'm not certain of that, but that's what seems to be happening in my case, which is of limited use. The purge code is within the controller's actionIndex.

**Comment by j-greig** [comment]
Hello Verbb, is there a roadmap for addressing this issue?

We are experiencing CPU overload due to database calls on a client site and I'm wondering if the wishlist db size is part of the problem.

If there was a way to discard empty wishlists on a more regular (& automatic) basis I think that might help?

**Comment by engram-design** [comment]
Added in [1.1.0](https://github.com/verbb/wishlist/releases/tag/1.1.0), Purge lists has now been added to Craft's Garbage Collection service, so this should be done in a much more automated fashion now.

In addition, there's also a `purgeInactiveGuestListsDuration` config setting that by default is set to 1 day, so its a bit more aggressive.

**Comment by j-greig** [comment]
Hello @engram-design we still have millions of elements in our db which I think are left over from this issue, I was wondering if we could hire you to help us prune them out safely? (it is still causing us issues whenever syncing or cloning the db)

**Comment by dannylewin** [comment]
I noticed this yesterday when the elements table seemed abnormally large. I grouped the lists by IP and there were some with thousands of lists under each IP. I googled the worst culprit and turned out to be the Google Crawler. But there was also the Bing crawler in there too.

I'm using a slightly older version, so this may have been changed/fixed now (doesn't look like it has though).  But it's likely caused by lists being created automatically when you load the site. Including bots/crawlers.
